### PR TITLE
build: upgrade gradle usage to prep for 7.0

### PR DESCRIPTION
**Pull request by ericdeansanchez**
The following applies to Gradle 6.3 and this commit updates the current
Gradle version from **5.4** to **6.3**. NOTE: This commit DOES NOT
UPDATE the following,
```
sourceCompatibility = 1.6
targetCompatibility = 1.6
```
The idea behind the upgrade is to:
1. Stay up to date with the community
2. Comply with deprecation warnings
3. Prepare to automate our java build process (like we've done in Php)

The current build process yields deprecation warnings that will be
errors in Gradle 7.0. In anticipation of this, and in order to update
our build process to be reproduced machine-to-machine, this PR makes
the following changes:

- upgrade with the gradle wrapper to get better IDE support
  - `$ gradle wrapper --gradle-version 6.3 --distribution-type all`

**Deprecation Fixes**:
This style is deprecated...
```text
apply plugin: 'java'
apply plugin: 'maven'
```
In favor of:
```
plugins {
    --snip--
    id 'java'
    // Note: the use of 'maven' alone is deprecated
    // guidance is to use the following
    id 'maven-publish'
}
```
This style has been deprecated:
```
testCompile group: 'junit', name: 'junit', version:'4.12'
```
In favor of this:
```
testImplementation group: 'junit', name: 'junit', version:'4.12
```
- `classifier = 'javadoc'` is a deprecated feature
  - prefer `archiveClassifier.set("javadoc")`
- `classifier = 'sources'` is a deprecated feature
  - prefer `archiveClassifier.set("sources")`

There is more work to be done, but this is a start.

**The Following Pass**:
- [x] `$ gradle build`
- [x] `$ gradle test`
- [x] `$ gradlew build`
- [x] `$ gradlew test`

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Fri Jun 09 10:41:54 PDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
```
Comment (sherwinski) [inline gradle/wrapper/gradle-wrapper.properties:3]: In theory, do we risk breaking any end user setups by upgrading from `5.4` to `6.3`?

Comment (ericdeansanchez) [reply gradle/wrapper/gradle-wrapper.properties:3]: Does it break your setup? Can we rely on our test suite here or? What do you foresee breaking for our end users? Are you asking, "does Gradle 6.3 build for 1.6 in a way that's incompatible with the way Gradle 5.4 builds for 1.6?"

Comment (ericdeansanchez) [reply gradle/wrapper/gradle-wrapper.properties:3]: Phrased differently, I'd be concerned if this change breaks end user setups, while passing all tests and successfully completing all CI tasks. To fully answer you question, I'm not sure––anything is possible. However, by updating the way we build for release and by meticulously documenting the process, more light has been shed on the build process overall (I think). 

Could you offer line numbers in the `build.gradle` file that you think may affect us or our users?

It's always possible I missed something [here](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper)

**Review by sherwinski — APPROVED** [review]
Tested it locally and everything looks good 👍

**Pull request merged**